src/java.base/windows/classes/sun/nio/fs/WindowsPathParser.java

```diff
@@ -92,6 +99,18 @@ static Result parseNormalizedPath(String input) {
      *          Indicates if the path requires to be normalized
      */
     private static Result parse(String input, boolean requireToNormalize) {
+        // if a prefix is present, note its type and remove it
+        final int prefixType;
+        if (input.startsWith(LONG_PATH_UNC_PREFIX)) {
+            prefixType = PREFIX_TYPE_LONG_UNC;
+            input = "\\" + input.substring(7);
+        } else if (input.startsWith(LONG_PATH_PREFIX)) {
+            prefixType = PREFIX_TYPE_LONG;
+            input = input.substring(4);
+        } else {
+            prefixType = PREFIX_TYPE_NONE;
```

Comment:
This looks okay but it means 2 x startsWith tests for the common case. If the input doesn't start with LONG_PATH_PREFIX then there is no need to check if starts with the UNC version.

Another idea here is drop the PREFIX_TYPE_xxx fields and just have an expectedType field of type WindowsPathType.

Reply:
Fixed in 90ef2d30a9ed8f7ddb5891836663e5f8728a948d.